### DISCUSSION TITLE
How to do deadtime pwm in lua

### DISCUSSION THREAD

**Discussion by jokokiswan** [openLuat/LuatOS#90] [Q&A] [answered]
I just wonder how to drive transistor using deadtime pwm in luat, thank you.

**Comment by wendal** [comment]
not support yet

**Reply by jokokiswan** [reply]
thank you!

**Comment by MarivelSwickheimer6pv** [comment, marked as answer]
To drive a transistor using deadtime PWM in Lua, you can use a combination of timers and GPIO pins. Here's an example code snippet:

lua
![image](https://user-images.githubusercontent.com/127706356/224592878-97272389-3f81-47c6-bc93-6a902121abd4.png)
In this example, channel1 and channel2 are the GPIO pins connected to the transistor. frequency is the PWM frequency, dutyCycle1 is the duty cycle for channel1, and dutyCycle2 is the duty cycle for channel2.

The tmr.alarm function sets up a timer that will start channel2 after a specified deadTime. This ensures that there is a delay between turning off channel1 and turning on channel2, which is necessary to prevent a short circuit.

I hope this helps! Let me know if you have any further questions.

**Reply by jokokiswan** [reply]
Did you ever encounter vtask delay glitch? I cannot do logging on my latest luat also. Like on this discussion: https://github.com/openLuat/LuatOS/discussions/92
